hydroflow/examples/kvs_bench/buffer_pool/serialization.rs

```diff
@@ -0,0 +1,68 @@
+use super::{AutoReturnBuffer, BufferPool};
+use crate::buffer_pool::{AutoReturnBufferInner, BufferType};
+use serde::{
+    de::{DeserializeSeed, Visitor},
+    Serialize, Serializer,
+};
+use std::{
+    cell::RefCell,
+    rc::{Rc, Weak},
+};
+
+impl Serialize for AutoReturnBuffer {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_bytes(&*self.borrow().unwrap())
+    }
+}
+
+pub struct AutoReturnBufferDeserializer {
+    pub collector: Weak<RefCell<BufferPool>>,
+}
+
+impl<'de> DeserializeSeed<'de> for AutoReturnBufferDeserializer {
+    type Value = AutoReturnBuffer;
+
+    fn deserialize<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct BytesVisitor;
+        impl<'de> Visitor<'de> for BytesVisitor {
+            type Value = BufferType;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("[u8; _]")
```

Comment:
expecting `"[u8; 1024]"` if I understand correctly

Reply:
Ok actually you can use `std::any::type_name::<BufferType>()` to future-proof it